src/defaults/commandEdit.tsx

```diff
@@ -0,0 +1,122 @@
+import * as ReactDOM from 'react-dom';
+import * as React from 'react';
+
+import { JupyterLabPlugin, JupyterLab } from '@jupyterlab/application';
+
+import {
+    INotebookTracker,
+    Notebook,
+    NotebookPanel
+} from '@jupyterlab/notebook';
+
+import { IDefaultStatusesManager } from './manager';
+
+import { Widget } from '@phosphor/widgets';
+
+export namespace CommandEditComponent {
+    export interface IState {
+        mode: string;
+    }
+    export interface IProps {
+        tracker: INotebookTracker;
+    }
+}
+
+export class CommandEditComponent extends React.Component<
+    CommandEditComponent.IProps,
+    CommandEditComponent.IState
+> {
+    state = {
+        mode: ''
+    };
+    constructor(props: CommandEditComponent.IProps) {
+        super(props);
+        this.props.tracker.currentChanged.connect(this.notebookChanged);
+    }
+
+    notebookChanged = (tracker: INotebookTracker, panel: NotebookPanel) => {
+        if (panel) {
+            this.setState({
+                mode: panel.content.mode
+            });
+            panel.content.stateChanged.connect(this.notebookStateChanged);
+        }
+    };
+
+    notebookStateChanged = (notebook: Notebook) => {
+        this.setState({
+            mode: notebook.mode
+        });
+    };
+    // onClick = {this.handleClick}
+
+    handleClick = () => {
```

Comment:
Also, instead of rolling our own interaction, possibly we should look for Commands that cover this same function, and use those instead. I know there is a `notebook:enter-command-mode` command that you could possibly use.